File: Sorting/InsertionSort.cs
using System;

public class Program
{
    public static void Main(string[] args)
    {
        int[] arr = { 13, 46, 24, 52, 20, 9 };
        
        InsertionSorting(arr);
        
        Console.WriteLine("Sorted array:");
        foreach (int num in arr)
        {
            Console.Write(num + " ");
        }
    }

    static void InsertionSorting(int[] arr)
    {
        int n = arr.Length;
        for (int i = 1; i < n; i++)  
        {
            int j = i;
            while (j > 0 && arr[j - 1] > arr[j])
            {
                int temp = arr[j - 1];
                arr[j - 1] = arr[j];
                arr[j] = temp;
                j--;
            }
        }
    }
}

File: Sorting/insertionSort.js
const insertionSort = (arr) => {
  let n = arr.length;
  for (let i = 1; i < n; i++) {  
    let j = i;
    while (j > 0 && arr[j - 1] > arr[j]) {
      let temp = arr[j - 1];
      arr[j - 1] = arr[j];
      arr[j] = temp;
      j--;
    }
  }
  return arr;
};

let arr = [13, 46, 24, 52, 20, 9];
console.log(insertionSort(arr));
